filebeat/harvester/reader/timeout.go

```diff
@@ -58,6 +60,8 @@ func (p *Timeout) Next() (Message, error) {
 	}
 
 	select {
+	case <-p.done:
+		return Message{}, ErrReaderStopped
```

Comment:
with error code being forwarded from `p.reader`, I wonder if we need the channel here.